spec/services/declarations/mark_as_payable_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+RSpec.describe Declarations::MarkAsPayable do
+  let(:application) { create(:application, :eligible_for_funded_place) }
+  let(:lead_provider) { application.lead_provider }
+  let(:statement) { create(:statement, :next_output_fee, lead_provider:) }
+  let(:declaration) { create(:declaration, :submitted_or_eligible, lead_provider:, application:) }
+  let!(:statement_item) { create(:statement_item, :eligible, statement:, declaration:) }
+
+  subject { described_class.new(statement:) }
+
+  describe "#mark" do
+    it "transitions the declaration state itself" do
+      expect {
+        subject.mark(declaration:)
+        declaration.reload
+      }.to change(declaration, :state).from("eligible").to("payable")
+    end
+
+    it "transitions the correct statement item state" do
```

Comment:
Worth adding in a statement item on the statement not associated to this declaration to ensure it doesn't get changed?